CLDR-15405 nameOrder→nameOrderLocales, mv sorting from usage→order, sampleNames in root / ∅∅∅ in en

pedberg-icu opened the pull request:
CLDR-15405

- [ ] This PR completes the ticket.

<!--
Thank you for your pull request.
Please see http://cldr.unicode.org/index/process for general
information on contributing to CLDR.

1. Make sure the ticket is filed at
https://unicode-org.atlassian.net/projects/CLDR/
2. Update the PR title and first line of this
message to include the ticket ID (CLDR-_____)
3. You will be automatically asked to sign the contributors’
license before the PR is accepted.
- sign: https://cla-assistant.io/unicode-org/cldr
- license: http://www.unicode.org/copyright.html#License
-->

Various items to unblock other work:
1. Change nameOrder to nameOrderLocales, make order attribute distinguishing, move locale list from attribute to value. Use root/en data suggested by Mark.
2. Move sorting from usage to order. Rearrange root/en data to match.
3. For sampleName, expanded & renamed items types, included givenInformal. Added root data with all nameFields for each item; adjusted the en data with ∅∅∅ for unused fields (add PathDescriotion note for that). The en sampleName data needs better values, current ones are just to have something for testing 
4. Misc en updates. remove . in mononym patterns since it is in initialPattern; use -initial-allCaps for all mononym patterns. 
5. Use PathHeader ordering functions for personName/sampleName section heads too.

Still to do (another PR): 
6. PlaceHolderDescriptions for non-{0},{1}-type placeholders
7. Better en sampleName data

macchiati commented on common/main/en.xml:
```diff
@@ -9129,145 +9129,212 @@ annotations.
+			<nameField type="prefix">∅∅∅</nameField>
 			<nameField type="given">Katherine</nameField>
+			<nameField type="given2">∅∅∅</nameField>
+			<nameField type="givenInformal">∅∅∅</nameField>
```
this needs to be given-informal

macchiati replied:
and elsewhere.

pedberg-icu replied:
I thought we said that we wanted attribute values to be camelCase (as opposed to the placeholder values, which can have appended hyper-separated modifiers)? Of course this means a mapping in the enums between the attribute value and the placeholder value).

macchiati commented on tools/cldr-code/src/main/java/org/unicode/cldr/util/PathHeader.java:
```diff
@@ -1878,8 +1878,8 @@ public String transform(String source) {
-                    final List<String> sampleItemValues = Arrays.asList("minimal", "simple", "full", "multiword");
-                    final List<String> fieldTypeValues = Arrays.asList("prefix", "given", "given2", "surname", "surname2", "suffix");
+                    final List<String> sampleItemValues = Arrays.asList("givenSurname", "given2Surname", "givenSurname2", "informal", "full", "multiword");
+                    final List<String> fieldTypeValues = Arrays.asList("prefix", "given", "given2", "givenInformal", "surname", "surname2", "suffix");
```
given-informal

macchiati commented on tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java:
```diff
@@ -1308,8 +1313,10 @@ public PersonNameFormatter(CLDRFile cldrFile) {
                 String value = cldrFile.getStringValue(path);
-                XPathParts parts = XPathParts.getFrozenInstance(path);
-                names.put(SampleType.valueOf(parts.getAttributeValue(-2, "item")), ModifiedField.from(parts.getAttributeValue(-1, "type")), value);
+                if (!value.equals("∅∅∅")) { // TODO is this how we want to handle ∅∅∅?
```
yes, that should work.

macchiati reviewed (changes requested):
Nice! I had only one substantive comment; use of -informal

pedberg-icu commented on tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java:
```diff
@@ -208,6 +210,9 @@ public ModifiedField(Field field, Modifier... modifiers) {
         public static ModifiedField from(String string) {
+            if (string.equals("givenInformal")) {
+                string = "given-informal"; // TODO: hack for handling nameField types vs namePattern values
+            }
```
This is where I currently map from the attribute value to the placeholder value

macchiati commented:
No, it uses the same syntax. It is only *within* a single field or modifier
(eg allCaps) that we use 🐪 instead of -

On Fri, Apr 1, 2022, 21:46 Peter Edberg ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In common/main/en.xml
> <https://github.com/unicode-org/cldr/pull/1869#discussion_r841007706>:
>
> >  			<nameField type="given">Katherine</nameField>
>
> +			<nameField type="given2">∅∅∅</nameField>
>
> +			<nameField type="givenInformal">∅∅∅</nameField>
>
>
> I thought we said that we wanted attribute values to be camelCase (as
> opposed to the placeholder values, which can have appended hyper-separated
> modifiers)? Of course this means a mapping in the enums between the
> attribute value and the placeholder value).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/1869#discussion_r841007706>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMC3PLZ4HLZB5XT4PK3VC7GITANCNFSM5SKBLY6Q>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

macchiati commented on common/main/en.xml:
```diff
@@ -9129,145 +9129,212 @@ annotations.
+			<nameField type="prefix">∅∅∅</nameField>
 			<nameField type="given">Katherine</nameField>
+			<nameField type="given2">∅∅∅</nameField>
+			<nameField type="givenInformal">∅∅∅</nameField>
```
```suggestion
			<nameField type="given-informal">∅∅∅</nameField>
```

macchiati replied:
Hmmm, I didn't even know there was an "add suggestion"

pedberg-icu commented:
> No, it uses the same syntax. It is only *within* a single field or modifier (eg allCaps) that we use 🐪 instead of -

OK, will fix. Then for ordering of the nameField type attribute values I will treat modifiers like "-informal" as less significant than the base names they modify.

macchiati commented:
Sounds good, thanks.

Note: if it would make the code cleaner to put ordering utilities into
PersonNameFormatter we could do that.

On Sat, Apr 2, 2022 at 11:03 AM Peter Edberg ***@***.***>
wrote:

> No, it uses the same syntax. It is only *within* a single field or
> modifier (eg allCaps) that we use 🐪 instead of -
>
> OK, will fix. Then for ordering of the nameField type attribute values I
> will treat modifiers like "-informal" as less significant than the base
> names they modify.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/1869#issuecomment-1086692275>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMBGUWKS262CW3YR6YLVDCDVPANCNFSM5SKBLY6Q>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

pedberg-icu commented:
> Sounds good, thanks. Note: if it would make the code cleaner to put ordering utilities into PersonNameFormatter we could do that.

I think better to have them in PathHeader.java, that is the main place they are relevant (I think)

pedberg-icu commented:
OK, changed nameField type attribute value givenInformal → given-informal (like the namePattern placeholder), adjusted OathHeader sampleNameOrder function accordingly, and squashed. @macchiati, can you re-review? Thanks.
